dashboard/test/models/course_offering_test.rb

```diff
@@ -168,6 +203,411 @@ class CourseOfferingTest < ActiveSupport::TestCase
+    unit2 = create(:script, name: 'unit2', family_name: 'family-8', version_year: '1992', is_course: true, published_state: 'in_development')
+    CourseOffering.add_course_offering(unit2)
+
+    assert unit1.course_version.course_offering.any_versions_in_development?
+  end
+
+  test 'any_versions_in_development? is false if none of the course versions are in development' do
+    unit1 = create(:script, name: 'unit1', family_name: 'family-9', version_year: '1991', is_course: true, published_state: 'beta')
+    CourseOffering.add_course_offering(unit1)
+    unit2 = create(:script, name: 'unit2', family_name: 'family-9', version_year: '1992', is_course: true, published_state: 'beta')
+    CourseOffering.add_course_offering(unit2)
+
+    refute unit1.course_version.course_offering.any_versions_in_development?
+  end
+
+  test 'assignable? is false if can not be instructor' do
+    refute @unit_teacher_to_students.course_version.course_offering.assignable?(@student)
+    assert @unit_teacher_to_students.course_version.course_offering.assignable?(@teacher)
+
+    refute @unit_facilitator_to_teacher.course_version.course_offering.assignable?(@teacher)
+    assert @unit_facilitator_to_teacher.course_version.course_offering.assignable?(@facilitator)
+  end
+
+  test 'assignable? is true if has pilot access and any course version is in pilot state' do
+    refute @pilot_unit.course_version.course_offering.assignable?(@teacher)
+    assert @pilot_unit.course_version.course_offering.assignable?(@pilot_teacher)
+    refute @pilot_unit.course_version.course_offering.assignable?(@pilot_instructor)
+
+    refute @pilot_pl_unit.course_version.course_offering.assignable?(@teacher)
+    refute @pilot_pl_unit.course_version.course_offering.assignable?(@pilot_teacher)
+    assert @pilot_pl_unit.course_version.course_offering.assignable?(@pilot_instructor)
+  end
+
+  test 'assignable? is true if any versions in development and user is levelbuilder' do
+    unit1 = create(:script, name: 'unit2', family_name: 'family-10', version_year: '1992', is_course: true, published_state: 'in_development')
+    CourseOffering.add_course_offering(unit1)
+
+    refute unit1.course_version.course_offering.assignable?(@teacher)
+    assert unit1.course_version.course_offering.assignable?(@levelbuilder)
+  end
+
+  test 'get assignable pl course offerings for teacher should return no offerings' do
+    assert_equal CourseOffering.assignable_pl_course_offerings(@teacher).length, 0
+  end
+
+  test 'get assignable course offerings for student should return no offerings' do
+    assert_equal CourseOffering.assignable_course_offerings(@student).length, 0
+  end
+
+  test 'get assignable student course offerings for student should return no offerings' do
+    assert_equal CourseOffering.assignable_student_course_offerings(@student).length, 0
+  end
+
+  test 'get assignable course offerings for teacher should return offerings where teacher can be instructor' do
```

Comment:
these tests looks brittle -- any minor change to the `assignable_course_offerings_info` method will require updating the tests in many places. because they each validate every detail, it's hard to see what the test is doing by reading the code, and it will be hard to see what is wrong in the case of a unit test failure.

 instead I would focus on testing specific business logic. the name of this test case is already a great statement of business logic. to verify this statement, I would argue that you only need to check the names of the unit groups, course versions and units.

if you want to get coverage of more of the fields, I would first challenge you to write more test case names oriented around the business logic you are testing, then write the minimum amount of test to validate that logic.

if you still want a test that covers every detail of the output, then I would recommend a minimum number of tests to validate the format of that output, rather than replicating the full `expected_course_offering_info` across many test cases. e.g. "assignable_course_offerings_info produces complete summary of course_versions and units", though the value of such a test is still a bit murky to me.

Reply:
Thanks for this feedback. I will revisit the tests and update you once I have improved them.

Reply:
Alright improvements have been made

Reply:
Looks great, thank you Dani!